std/experimental/allocator/building_blocks/kernighan_ritchie.d

```diff
@@ -525,6 +530,15 @@ struct KRRegion(ParentAllocator = NullAllocator)
         return null;
     }
 
+    private void ensureCorrectSize()
+    {
+        import std.math : abs;
+
+        if (root && (abs(payload.length - root.size) < Node.sizeof)) {
```

Comment:
I wanted to know if fixing them would indeed make a difference and needed a test case to highlight their effect. So I did a bit of digging.

Interesting fact (explained only for 64 bit). This bug is influenced by the size of the buffer and the alloc sizes.
We know that the alignment on 64 bit systems is 8 and ```goodAllocSize``` returns multiples of the alignment: ``` roundUpToMultipleOf(alignment); ```

Take for example the buffer in the bug report with a size of 256 * 1024. This value is a multiple of both 8 and 16.  ```KRRegion``` allocates in chunks with sizes multiples of 8, but our minimum alloc size is 16. If we allocate such values that there is an odd number of multiples of 8, but not multiple of 16, then we will have those leftover bytes if the last amount allocated is a multiple of 16. They are the difference between the multiple of 8 and the multiple of 16 - always 8.

In the ```unittest``` the values are already rounded: [18904, 2008, 74904, 224, 111904, 1904, 52288] there are 3 multiples of 8 at the beginning which are not multiples of 16 => 2008, 74904 and 18904.
PS. KRRegion payloads with a size !=multiple of 8 behave the same, only that the leftover bytes (<8) could never be used for allocation.

So at this point I can actually create a situation where such a gap appears between 2 nodes and not just at the end of the memory/buffer/payload.
Have this allocation order :```[2008, 18904, 74904, 224, 111904, 1904, 52288]```
deallocate(18904)
allocate(18896) <-- 18904 - 8
Gap created between 2nd and 3rd block!

This would indeed justify fixing ```Node.adjacent``` and ```Node.coalesce```. And we also have a test to prove it now. :)